optimum/intel/launcher.py

```diff
@@ -0,0 +1,14 @@
+from neural_coder.launcher import Launcher
+
+from .neural_compressor.neural_coder_adaptor import NeuralCoderAdaptor
```

Comment:
This file should be moved to the `neural_compressor` directory as it's related to INC.
Also the command line `python -m optimum.intel.neural_compressor.launcher run_glue.py` is better for clarity I think, as the optimizations are related to INC

Reply:
NeuralCoder as a launcher provides the capability to expand to the other use cases e.g., IPEX, OV. So that's why we intend to make this under intel. Does it make sense?

Reply:
We have been discussing this: the reason we currently put ```launcher.py``` at this location (under ```intel```) is because Neural Coder in the future might enable more Intel optimizations for Optimum-Intel like IPEX etc, which means it is not limited to INC. Also we're thinking if the command line becomes ```optimum.intel.neural_compressor.launcher``` it seems a bit long and less user-friendly. @hshen14 Any additional comments?

Reply:
Because of the `neural_coder` dependency, I think it would make more sense to first add it to `neural_compressor` and to move it in the future in case of needs, what do you think ?

Reply:
hi @echarlaix I have made the change per our discussion